### ISSUE TITLE
could the date of the post be added in the search results?

### ISSUE THREAD

**Issue by pamag** [Bioconductor/legacy.support.bioconductor.org#12]
I would find very useful to have the date of the post in the search results page. Is that possible?
When dealing with software I want to prioritize the answers by date mainly because  I want to read the questions or answers related to the version I have or at least with the current paramteres or options. A three years old answer probably would not have the right options or paramaters or will mention deprecated solutions anyway, so it would not be my first option.

**Comment by nhayden** [comment]
:+1:

**Comment by dtenenba** [comment]
if I'm not mistaken this is one of the things that @ialbert is going to work on.

**Comment by ialbert** [comment]
yes, this feature can be added and I will make a note of it. New version will go into beta testing this week most likely.

**Comment by ialbert** [comment]
The feature has been added in 3.0 

https://github.com/ialbert/biostar-central/commit/27ba9e736d9f989b2874108a279eeddd9af2396e

**Comment by dtenenba** [comment]
Thanks. Can you tell me a little bit about 3.0? Is it a beta? When will it be merged with master? And what do I need to do to upgrade to it? The usual merging, but will there be any migrations involved and how do I deal with that? Thanks again for the fix!

**Comment by ialbert** [comment]
The main coding is essentially done for 3.0

There will be a process of testing and bugfixing for a few weeks.

The system will auto migrate itself, you just need to check out the new
code, point it to the same database (or a copy of it) and apply the
migration command. I tried to build into the new version all the features
that I saw you guys add into the current site. You will be able to merge
into your master but it needs to be done so that it accepts all upstream
changes.

The editor has been swapped to markdown, email integration is better and
overall the code is a whole lot simpler to modify and edit. One major
feature is adding groups, these are isolated parts of the site that live
under a subdomain of the main site that only show posts made into that
subdomain for example:

developers.support.bioconductor.org vs support.bioconductor.org will show
different posts, users etc.

**Comment by dtenenba** [comment]
Sounds great. I will probably test it on our docker container for the support site but refrain from going live with the changes until you are done with testing and bugfixing--can you let me know when you merge?

The groups feature sounds cool. It's slightly tricky for us to add subdomains as any DNS change involves  making the change externally but also asking for it to be made internally because inside FHCRC the DNS queries for bioconductor.org are resolved by an internal DNS server. Also, topic.support.bioconductor.org is a little unwieldy. Also I think I'd have to make a change in apache for each subdomain added.

Is there a way to make groups work on a url basis instead?

Like 

support.bioconductor.org/groups/developers/bla

where bla is whatever would normally be on the end of the url?

That would also make it easier to test out on our docker container, as I wouldn't have to figure out how to make DNS work on it.

Let me know if that makes any sense....

**Comment by ialbert** [comment]
think of a group as it were another biostar site. The entire site may look
different and have different content, not just the post list, hence it is
not as simple as just adding an extra url.

www.site.com/user/list
www.site.com/post/list
www.site.com/help

will produce different results than

devs.site.com/user/list
devs.site.com/post/list
devs.site.com/help

yet is still served/indexed/seached by the one installation. Users can join
and follow any of the groups in different ways, some by messages other via
email etc. some groups may be private and only visible to users etc.

The DNS part is actually easy to set up, one just needs a catch all pattern
that redirects all subdomains to resolve onto the site itself: *.site.com
-> site.com the code in Biostar will handle the rest. Also the feature is
purely optional, if you don't allow groups none will created - but I am
quite sure there will be nice use cases.

But it is true that the same could be achieved via inserting the group into
every url.

www.site.com/user/list/
www.site.com/devs/user/list

We'll see how it works in practice and could be changed to that

**Comment by mikelove** [comment]
hi, 

I know there isn't an abundance of spare time to develop the support site, but I have a hard time finding my own answers to point people to using the current support site search (likewise I imagine users give up quickly and end up posting what are actually duplicate questions). 

If the search results could be ranked by posting time that would be huge for me. I have a really hard time finding relevant threads where I've provided answers, even when those threads are only a month old.

**Comment by mikelove** [comment]
related threads: 

https://github.com/Bioconductor/support.bioconductor.org/issues/4
https://github.com/Bioconductor/support.bioconductor.org/issues/13

**Comment by ialbert** [comment]
Hi Mike,

this is one of those things that have a good news - bad news component. The good news is that search delegated to the [ElasticSearch](https://www.elastic.co/) engine and Biostar connects to that via an API - there is a minimal amount of code is necessary to integrate the two. This engine is  well documented and can be configured via a JSON like language and offers a very wide variety of search strategies for returning and sorting the hits. The engine is built such that it is controlled mainly via the configuration language rather than code.

The bad news is that I never had time to figure it all out - and is something that I'd like to investigate myself but things always seem to get in the way.

**Comment by mikelove** [comment]
Thanks @ialbert for your explanation and your work on the site!
